piksi_tools/fileio.py

```diff
@@ -474,7 +683,60 @@ def printable_text_from_device(data):
     return str
 
 
+def mk_progress_cb(file_length):
+
+    time_last = [time.time()]
```

Comment:
Since this is for the progress function we don't need to use `Time.now()` -- `Time.now()` is only special in that it only tracks seconds and milliseconds.